app/packages/core/src/components/ViewBar/ViewBar.tsx

```diff
@@ -83,21 +83,24 @@ const ViewBar = React.memo(() => {
       setView,
       fieldNames: fieldPaths,
     });
-  }, [view]);
+  }, [view, fieldPaths, send]);
```

Comment:
@kaixi-wang did we add these? I'd say let's hold off on adding dependencies to useEffects unless they fix a specific issue. useEffect has side effects and in general I like to test the entire app when dependencies get added to them. I remove them. lmk if you object

Reply:
the react hooks linter said to add them so I did and it seemed to make things more stable on my side... Also reading up on hooks, it sounded like we need to be adding them https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies...

Reply:
I agree with you and linter :) to keep the risks low (like the sample modal not opening in another thread in the pr) I simply suggested to keep dependency changes minimal unless they fix a specific scenario with saved-views